src/app/templates/TezosOperationView/tezos-transaction-view.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { memo, useCallback, useMemo } from 'react';
+
+import BigNumber from 'bignumber.js';
+import { nanoid } from 'nanoid';
+import { FormProvider } from 'react-hook-form-v7';
+
+import { toastError } from 'app/toaster';
+import { TEZ_TOKEN_SLUG } from 'lib/assets';
+import { TEZOS_BLOCK_DURATION } from 'lib/fixed-times';
+import { useTypedSWR } from 'lib/swr';
+import { mutezToTz, tzToMutez } from 'lib/temple/helpers';
+import { TempleTezosDAppOperationsPayload } from 'lib/temple/types';
+import { tezosManagerKeyHasManager } from 'lib/tezos';
+import { serializeError } from 'lib/utils/serialize-error';
+import { getAccountAddressForTezos } from 'temple/accounts';
+import { TezosChain, useAllAccounts, useAllTezosChains } from 'temple/front';
+import { StoredTezosNetwork } from 'temple/networks';
+import { getReadOnlyTezos } from 'temple/tezos';
+import { TempleChainKind } from 'temple/types';
+
+import { OperationViewLayout } from '../operation-view-layout';
+import { TezosTxParamsFormData } from '../TransactionTabs/types';
+import { useTezosEstimationForm } from '../TransactionTabs/use-tezos-estimation-form';
+
+export interface TezosTransactionViewProps {
+  payload: TempleTezosDAppOperationsPayload;
+  formId: string;
+  error: any;
+  setTotalFee: SyncFn<number>;
+  setStorageLimit: SyncFn<number>;
+  onSubmit: EmptyFn;
+}
+
+export const TezosTransactionView = memo<TezosTransactionViewProps>(
+  ({ payload, formId, error: submitError, setTotalFee, setStorageLimit, onSubmit }) => {
+    const { networkRpc, opParams, sourcePkh, estimates, error: estimationError } = payload;
+    const tezosChains = useAllTezosChains();
+    const accounts = useAllAccounts();
+    const sendingAccount = useMemo(
+      () => accounts.find(acc => getAccountAddressForTezos(acc) === sourcePkh)!,
+      [accounts, sourcePkh]
+    );
+
+    const estimationData = useMemo(() => {
+      if (!estimates) return undefined;
+
+      const revealFeeMutez = mutezToTz(estimates.length > opParams.length ? estimates[0].suggestedFeeMutez : 0);
+
+      return {
+        estimates,
+        baseFee: mutezToTz(BigNumber.sum(...estimates.map(est => est.suggestedFeeMutez + est.burnFeeMutez))),
+        gasFee: mutezToTz(BigNumber.sum(...estimates.map(est => est.suggestedFeeMutez))),
+        revealFee: revealFeeMutez
+      };
+    }, [estimates, opParams.length]);
+
+    const tezos = useMemo(() => getReadOnlyTezos(networkRpc), [networkRpc]);
+
+    const getTezosChain = useCallback(async (): Promise<TezosChain> => {
+      const knownTezosChain = Object.values(tezosChains).find(c =>
+        c.allRpcs.some(rpc => rpc.rpcBaseURL === networkRpc)
+      );
+
+      if (knownTezosChain) {
+        return knownTezosChain;
+      }
+
+      const chainId = await tezos.rpc.getChainId();
```

Comment:
what will happen if this rpc call fails?

Reply:
In this case, an error overlay will be displayed. If a dApp requests a transaction on an unknown RPC node which does not respond, there is nothing we can do with it